addons/web_tour/static/src/js/tour_manager.js

```diff
@@ -295,7 +305,9 @@ return core.Class.extend(mixins.EventDispatcherMixin, ServicesMixin, {
         if (this.running_tour !== tour_name) {
             tip.widget.on('tip_consumed', this, this._consume_tip.bind(this, tip, tour_name));
         }
-        tip.widget.attach_to($anchor).then(this._to_next_running_step.bind(this, tip, tour_name));
+        tip.widget.attach_to($anchor)
+            .then(this._to_next_running_step.bind(this, tip, tour_name))
+            .then(this.setTooltipReady);
```

Comment:
why did you need this?

Reply:
This was essentially to avoid calling needless `testUtils.nextTicks` in the tests, but it could be used in the future as an assertion that the tour manager and its widgets are fully ready.

Reply:
Everything should be sync. If it's not, I guess it's because of a settimeout. You can parameterize the timeout value and override it in the test like we often do.

Reply:
I don't really see a usecase where we would need this, and it makes the code a bit more complicated, so if we can avoid this, I think it's better.